services/venice-controller/src/main/java/com/linkedin/venice/controller/server/CreateVersion.java

```diff
@@ -792,7 +803,14 @@ public Route emptyPush(Admin admin) {
         responseObject.setKafkaTopic(version.kafkaTopicName());
         responseObject.setKafkaBootstrapServers(version.getPushStreamSourceAddress());
 
-        admin.writeEndOfPush(clusterName, storeName, versionNumber, true);
+        if (!previousVersions.contains(version)) {
```

Comment:
so we allow concurrent empty push with different pushId?

Reply:
No this is best-effort to prevent writing SOP/EOP. Concurrent push logic is unchanged meaning that we won't allow it

Reply:
how did we allow concurrent empty push in the test ?

Reply:
Actually this line returned, should we harden this to block concurrent batchpushes as well with same id?
`  Optional<Version> versionWithPushId = getVersionWithPushId(clusterName, storeName, pushJobId);
          if (versionWithPushId.isPresent()) {
            return new Pair<>(false, versionWithPushId.get());
          }`

Reply:
Two empty pushes with the same push job ID are not considered concurrent. The second push doesn’t create a new version; it simply ends up writing the SOP and EOP markers again.

Reply:
In the else part can we indicate its a duplicate push and send the response back to the user?

Reply:
We're sending response back to the user which is VersionCreationResponse.

Reply:
Added logs